### PR TITLE
Design space axis to IR axis

### PR DESCRIPTION
**Pull request by rsheeter**
Starting to rough in some of the patterns for ufo2ir. We are as yet somewhat far from doing anything useful :)

### CODE REVIEW

File: ir/src/ir.rs [hunk 1]
```diff
@@ -4,11 +4,11 @@ use serde::{Deserialize, Serialize};
 
 #[derive(Serialize, Deserialize, Debug, PartialEq)]
 pub struct Axis {
-    tag: String,
-    min: f32,
-    default: f32,
-    max: f32,
-    hidden: bool,
+    pub tag: String,
+    pub min: f32,
```
Comment (cmyr) [inline ir/src/ir.rs:8]: Is `f32` the most appropriate type here? Genuinely a question.

File: ufo2ir/src/error.rs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum UfoToIrError {
+    // It is in fact inconceivable that this would fail
```
Comment (cmyr) [inline ufo2ir/src/error.rs:5]: 👍

File: ufo2ir/src/toir.rs [hunk 3]
```diff
@@ -0,0 +1,46 @@
+use ir::ir;
+use norad::designspace::{self, DesignSpaceDocument};
+
+use crate::error::UfoToIrError;
+
+pub fn designspace_to_ir(designspace: DesignSpaceDocument) -> Result<Vec<ir::Axis>, UfoToIrError> {
```
Comment (cmyr) [inline ufo2ir/src/toir.rs:6]: So just an observation, but I could imagine using a trait to encapsulate "convertibility to IR".

**Review by cmyr — APPROVED** [review]
some observations inline, looks good!

File: ir/src/ir.rs (hunk 1 above)
Comment (rsheeter) [reply ir/src/ir.rs:8]: Good question. I don't want to block on or forget this so I've filed https://github.com/googlefonts/fontmake-rs/issues/6

**Pull request merged**